### ISSUE TITLE
add xaml support

### ISSUE THREAD

**Issue by furesoft** [Haydend/ConsoleDraw#1]
pls add xaml support or loading ui from xml

**Comment by Haydend** [comment]
I agree this would be a great feature. However, this would be a major undertaking and requires time and knowledge I don't currently have.

If anyone would like to take a crack at this, please go ahead! I'm happy to offer any assistance.

**Comment by Lucidiot** [comment]
[This project](https://www.codeproject.com/Articles/15086/Forms-for-Console-Apps) has some custom XML support by implementing [IXmlSerializable](https://msdn.microsoft.com/en-us/library/system.xml.serialization.ixmlserializable(v=vs.110).aspx) in its classes.

**Comment by jinek** [comment]
I'm working on very similar project which assumes full XAML suppotr with data binding, animations etc.
It's TUI implementation of Avalonia UI https://github.com/jinek/Consolonia
Please, consider supporting the project

![](https://raw.githubusercontent.com/jinek/Consolonia/main/showcase.gif)
